boba/parser.py

```diff
@@ -290,8 +290,46 @@ def _code_gen(self):
             + len(self.code_parser.get_decisions())
 
         self.wrangler.create_dir()
-        for idx, p in enumerate(paths):
-            self._code_gen_recur(p, 0, '', History(idx))
+
+        if decrecords is None:
+            for idx, p in enumerate(paths):
+                self._code_gen_recur(p, 0, '', History(idx))
+        elif path_filter is None:
+            for idx, p in enumerate(paths):
+                self._code_gen_recur(p, 0, '',  History(idx, '', decrecords, []))
+        else:
```

Comment:
Similarly you might want to encapsulate this chunk of code in a function that returns `valid_paths`. And here you'll only need a line to get the valid paths. We can probably get rid of the if-else block too (if it's possible to pass an empty decrecord to History and it will still work properly). Smaller functions that do simple things are much easier to read and maintain.